### PR TITLE
✨ cleanup use controller-runtime utilities for finalizers

### LINKED ISSUE #2198: Update Finalizer handling to use controller-runtime utilities
**Detailed Description**

Currently we frequently use the following pattern for adding Finalizers in many controllers:

```go
	if !util.Contains(mp.Finalizers, clusterv1.MachinePoolFinalizer) {
		mp.Finalizers = append(mp.Finalizers, clusterv1.MachinePoolFinalizer)
	}
```

and removing Finalizers:

```go
mp.ObjectMeta.Finalizers = util.Filter(mp.ObjectMeta.Finalizers, clusterv1.MachinePoolFinalizer)
```

This was because we adopted the pattern prior to controller-runtime providing utilities around managing Finalizers. Now that [controller-runtime provides utilities for this](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil), we should adopt those.

We should update the current handling of adding and removing finalizers with [AddFinalzer](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#AddFinalizer) and [RemoveFinalizer](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#RemoveFinalizer).

/priority important-longterm
/help
/good-first-issue

### PR DESCRIPTION
**Pull request by daxmc99**
**What this PR does / why we need it**: 
Standardizes finalizer utils around controller-runtime

**Which issue(s) this PR fixes** :
Fixes #2198

### CODE REVIEW

**Comment by detiber** [comment]
The e2e test failure is a known issue right now.

File: test/infrastructure/docker/controllers/dockermachine_controller.go [hunk 1]
```diff
@@ -153,9 +154,7 @@ func (r *DockerMachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 
 func (r *DockerMachineReconciler) reconcileNormal(ctx context.Context, machine *clusterv1.Machine, dockerMachine *infrav1.DockerMachine, externalMachine *docker.Machine, externalLoadBalancer *docker.LoadBalancer, log logr.Logger) (ctrl.Result, error) {
 	// If the DockerMachine doesn't have finalizer, add it.
-	if !util.Contains(dockerMachine.Finalizers, infrav1.MachineFinalizer) {
-		dockerMachine.Finalizers = append(dockerMachine.Finalizers, infrav1.MachineFinalizer)
-	}
+	controllerutil.AddFinalizer(dockerMachine,infrav1.MachineFinalizer)
```
Comment (ncdc) [inline test/infrastructure/docker/controllers/dockermachine_controller.go]:
```suggestion
	controllerutil.AddFinalizer(dockerMachine, infrav1.MachineFinalizer)
```

File: test/infrastructure/docker/controllers/dockermachine_controller.go [hunk 2]
```diff
@@ -239,8 +238,7 @@ func (r *DockerMachineReconciler) reconcileDelete(machine *clusterv1.Machine, do
 	}
 
 	// Machine is deleted so remove the finalizer.
-	dockerMachine.Finalizers = util.Filter(dockerMachine.Finalizers, infrav1.MachineFinalizer)
-
+	controllerutil.RemoveFinalizer(dockerMachine,infrav1.MachineFinalizer)
```
Comment (ncdc) [inline test/infrastructure/docker/controllers/dockermachine_controller.go]:
```suggestion
	controllerutil.RemoveFinalizer(dockerMachine, infrav1.MachineFinalizer)
```

**Pull request merged**